src/msg/async/AsyncConnection.cc

```diff
@@ -362,14 +362,21 @@ ssize_t AsyncConnection::_try_send(bool more)
                              << " remaining bytes " << outcoming_bl.length() << dendl;
 
   if (!open_write && is_queued()) {
-    center->create_file_event(sd, EVENT_WRITABLE, write_handler);
-    open_write = true;
+    if (center->in_thread()) {
```

Comment:
sure.

The main idea is we want to remove file_lock in EventCenter, so each event thread which own EventCenter must operate EventCenter in local thread. And it means other threads can't operate others EventCenter and we must move EventCenter operations from other threads into local event thread.

I'm squash commit message into commit at last